code/DocGenerator.php

```diff
@@ -20,19 +20,47 @@ class DocGenerator extends Object implements Flushable
     /**
      * After the request, generate the docs.
      *
-     * @inheritdoc
+     * {@inheritdoc}
      */
     public static function flush()
     {
-        if (Config::inst()->get('DocGenerator', 'enabled') === true) {
-            $modules = Config::inst()->get('DocGenerator', 'document_modules');
-            foreach ($modules as $module => $location) {
-                /** @var Director $director */
-                $director = Injector::inst()->get('Director');
-                exec($director->baseFolder() . '/vendor/apigen/apigen/bin/apigen generate -q -s ' . $director->baseFolder() . "/$module -d " . $director->baseFolder() . "/$location/$module --exclude=*tests* --todo ");
-                DB::alteration_message('(Re)generated docs in ' . $director->baseFolder() . '/' . $location . '/' . $module . ' for ' . $module, 'good');
-            }
+        if ((Config::inst()->get('DocGenerator', 'enabled') === true) && (Config::inst()->get('DocGenerator', 'multi_module_enabled') === true)) {
+            self::processMulti();
+        } elseif (Config::inst()->get('DocGenerator', 'enabled') === true) {
+            self::processSingle();
         }
     }
 
+    /**
+     * Generate docs for folder/module.
+     */
+    private static function processSingle()
+    {
+        $director = Injector::inst()->get('Director');
+        $modules = Config::inst()->get('DocGenerator', 'document_modules');
+        foreach ($modules as $module => $location) {
+            exec($director->baseFolder().'/vendor/apigen/apigen/bin/apigen generate -q -s '.$director->baseFolder()."/$module -d ".$director->baseFolder()."/$location/$module --exclude=*tests* --todo ");
+            DB::alteration_message('(Re)generated docs in '.$director->baseFolder().'/'.$location.'/'.$module.' for '.$module, 'good');
+        }
+    }
+
+    /**
+     * Generate single docs from multiple folders/modules.
+     */
+    private static function processMulti()
+    {
+        $director = Injector::inst()->get('Director');
```

Comment:
Could you update all strings like `'ClassName'` to use `ClassName::class` without the quotes? That would make everything SS4 compatible in almost a single go.